[CI] SplitIndexIT fails with Lucene-7.4 snapshot

This test starts failing more often after upgrading to Lucene-7.4 snapshot. Most of the failures are on Windows. It may relate to this Lucene failure https://elasticsearch-ci.elastic.co/job/apache+lucene-solr+master/27765/consoleText.

Error: `still has pending deleted files; cannot initialize IndexWriter`

```
  1> Caused by: java.lang.IllegalArgumentException: Directory store(MMapDirectory@C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\server\build\testrun\integTest\J1\temp\org.elasticsearch.action.admin.indices.create.SplitIndexIT_734B52BB31887C24-001\tempDir-002\data\nodes\1\indices\Ru5mKdhxQpaT3gWmSTvH2w\16\index lockFactory=org.apache.lucene.store.NativeFSLockFactory@66502081) still has pending deleted files; cannot initialize IndexWriter
  1> 	at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:698) ~[lucene-core-7.4.0-snapshot-1ed95c097b.jar:7.4.0-snapshot-1ed95c097b 1ed95c097b82ee5f175e93f3fe62572abe064da6 - jimczi - 2018-05-03 16:16:13]
  1> 	at org.elasticsearch.index.store.Store.newIndexWriter(Store.java:1635) ~[main/:?]
  1> 	at org.elasticsearch.index.store.Store.bootstrapNewHistory(Store.java:1487) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.internalRecoverFromStore(StoreRecovery.java:393) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.lambda$recoverFromLocalShards$3(StoreRecovery.java:133) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.executeRecovery(StoreRecovery.java:301) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.StoreRecovery.recoverFromLocalShards(StoreRecovery.java:123) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.IndexShard.recoverFromLocalShards(IndexShard.java:1520) ~[main/:?]
  1> 	at org.elasticsearch.index.shard.IndexShard.lambda$startRecovery$8(IndexShard.java:2042) ~[main/:?]
  1> 	... 4 more
```

Some CI instances:

1. https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/1608/consoleText

2. https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/1609/console

Log: 
[shrink.txt](https://github.com/elastic/elasticsearch/files/1978298/shrink.txt)

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/indices.split/20_source_mapping.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/indices.split/20_source_mapping.yml
@@ -1,11 +1,8 @@
 ---
 "Split index ignores target template mapping":
   - skip:
-      # when re-enabling uncomment the below skips
-      version: "all"
-      reason: "AwaitsFix'ing, see https://github.com/elastic/elasticsearch/issues/30503"
-      # version: " - 6.3.99"
-      # reason: expects warnings that pre-6.4.0 will not send
+      version: " - 6.3.99"
+      reason: expects warnings that pre-6.4.0 will not send
       features: "warnings"
 
   # create index

--- a/server/src/main/java/org/elasticsearch/index/store/Store.java
+++ b/server/src/main/java/org/elasticsearch/index/store/Store.java
@@ -731,13 +731,13 @@ public class Store extends AbstractIndexShardComponent implements Closeable, Ref
 
         private final Logger deletesLogger;
 
-        StoreDirectory(Directory delegateDirectory, Logger deletesLogger) throws IOException {
+        StoreDirectory(Directory delegateDirectory, Logger deletesLogger) {
             super(delegateDirectory);
             this.deletesLogger = deletesLogger;
         }
 
         @Override
-        public void close() throws IOException {
+        public void close() {
             assert false : "Nobody should close this directory except of the Store itself";
         }
 
@@ -758,6 +758,17 @@ public class Store extends AbstractIndexShardComponent implements Closeable, Ref
         @Override
         public String toString() {
             return "store(" + in.toString() + ")";
+        }
+
+        @Override
+        public boolean checkPendingDeletions() throws IOException {
+            if (super.checkPendingDeletions()) {
+                deletesLogger.warn("directory has still pending deletes");
+            }
+            // we skip this check since our IW usage always goes forward.
+            // we still might run into situations where we have pending deletes ie. in shrink / split case
+            // and that will cause issues on windows since we open multiple IW instance one after another during the split/shrink recovery
+            return false;
         }
     }
 

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java
@@ -23,7 +23,6 @@ import org.apache.lucene.search.Sort;
 import org.apache.lucene.search.SortField;
 import org.apache.lucene.search.SortedSetSelector;
 import org.apache.lucene.search.SortedSetSortField;
-import org.apache.lucene.util.LuceneTestCase.AwaitsFix;
 import org.elasticsearch.Version;
 import org.elasticsearch.action.admin.cluster.reroute.ClusterRerouteResponse;
 import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest;
@@ -77,7 +76,6 @@ import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 
-@AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/30416")
 public class ShrinkIndexIT extends ESIntegTestCase {
 
     @Override

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/create/SplitIndexIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/create/SplitIndexIT.java
@@ -24,7 +24,6 @@ import org.apache.lucene.search.SortField;
 import org.apache.lucene.search.SortedSetSelector;
 import org.apache.lucene.search.SortedSetSortField;
 import org.apache.lucene.search.join.ScoreMode;
-import org.apache.lucene.util.LuceneTestCase;
 import org.elasticsearch.Version;
 import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest;
 import org.elasticsearch.action.admin.cluster.state.ClusterStateResponse;
@@ -80,7 +79,6 @@ import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 
 
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/30416")
 public class SplitIndexIT extends ESIntegTestCase {
 
     @Override

